core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -764,10 +769,13 @@ public void onUpdate(List<ResolvedServerInfoGroup> servers, Attributes config) {
         onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
         return;
       }
+      log.log(Level.FINE, "[{0}] resolved address: {1}, config={2}",
+          new Object[] {getLogId(), servers, config});
 
       try {
         balancer.handleResolvedAddresses(servers, config);
       } catch (Throwable e) {
+        log.log(Level.WARNING, "[" + getLogId() + "] Caught exception from LoadBalancer", e);
```

Comment:
No, it's only logged if there is a bug in `LoadBalancer`, because it's not expected to throw.

Here we catch and pass the exception back to `LoadBalancer`, in hope that it will be propagated to the application by LoadBalancer. However, LoadBalancer may not propagate or log as we expect, or there may not even be RPC yet to be propagated to, so we need to log here as the last resort.

We don't catch in `onError()` below, because if `handleNameResolutionError()` throws, there isn't a really good way to propagate this error to application. The exception will be caught by transport thread (or `ChannelExecutor` in v2 impl) and logged there.